### PR TITLE
Report json seed data

### PR DESCRIPTION
**Pull request by perrymant**
Created seed JSON file that is read:
1. in a test `FileReaderTest.java` using TDD.
2. in `Application.java` - this will be where the actual implementation of reading the JSON file occurs.

### CODE REVIEW

File: src/test/resources/report_seed_data_test.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (kinbiko) [inline src/test/resources/report_seed_data_test.json:1]: Why are there two JSON files? I'd expect only one

File: src/main/java/io/github/perrymant/moneymaker/Application.java [hunk 2]
```diff
@@ -1,9 +1,15 @@
 package io.github.perrymant.moneymaker;
 
+import java.util.ArrayList;
+import java.util.List;
+
 class Application {
     static final String ERROR_MESSAGE = "Error: Invalid argument: moneymaker help for more info.";
     private final Budget budget;
     private final FileReader reader = new FileReader();
+    private final List<Transaction> reportJSON = new ArrayList<>();
```
Comment (kinbiko) [inline src/main/java/io/github/perrymant/moneymaker/Application.java:10]: This, and the imports above do nothing. Let's delete it.

File: src/main/java/io/github/perrymant/moneymaker/Application.java [hunk 3]
```diff
@@ -18,6 +24,8 @@ void start(String[] args) {
                 logger.info(reader.read("moneymaker-help.txt"));
             } else if ("report".equals(args[0])) {
                 logger.info(budget.report());
+            } else if ("reportJSON".equals(args[0])) {
```
Comment (kinbiko) [inline src/main/java/io/github/perrymant/moneymaker/Application.java:27]: I don't understand what this is. Let's stick to the engineering design and task breakdown unless there's a really good reason not to.

Just to clarify: I'd expect #34 to result in a PR with literally only one file, and #35 to *replace* `report`, not add another.

File: src/test/java/io/github/perrymant/moneymaker/FileReaderTest.java [hunk 4]
```diff
@@ -10,6 +10,30 @@
     public void canDisplayHelpInfo() {
         assertEquals("Some Text", new FileReader().read("flatfile.txt"));
     }
+
+    @Test
+    public void canDisplay_report_seed_data_json() {
```
Comment (kinbiko) [inline src/test/java/io/github/perrymant/moneymaker/FileReaderTest.java:15]: We've already got this test case covered in the test above. No need to add this one.

**Review by kinbiko — CHANGES_REQUESTED** [review]
It might help me understand better what we're doing if you link to the issue this PR corresponds to. I've reviewed it as if it was addressing #34.

File: src/main/resources/report_seed_data.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    {
+      "date":"2018-01-01",
+      "amount":125,
+      "balance":125,
+      "description":"Got paid"
+    },
+    {
+      "date":"2018-01-02",
+      "amount":-72,
+      "balance":53,
```
Comment (kinbiko) [inline src/main/resources/report_seed_data.json:12]: I thought we're aiming to calculate this balance when we read the document, no?

File: src/test/resources/report_seed_data_test.json (hunk 1 above)
Comment (perrymant) [reply src/test/resources/report_seed_data_test.json:1]: One for the test, and one for the implementation - but will remove the test one.

File: src/main/java/io/github/perrymant/moneymaker/Application.java (hunk 2 above)
Comment (perrymant) [reply src/main/java/io/github/perrymant/moneymaker/Application.java:10]: Yep - sorry, this was me starting on the next sprint task... will remove.

File: src/main/java/io/github/perrymant/moneymaker/Application.java (hunk 3 above)
Comment (perrymant) [reply src/main/java/io/github/perrymant/moneymaker/Application.java:27]: Yep - just me trying it out here. Will replace it in the actual implementation!

File: src/test/java/io/github/perrymant/moneymaker/FileReaderTest.java (hunk 4 above)
Comment (perrymant) [reply src/test/java/io/github/perrymant/moneymaker/FileReaderTest.java:15]: Will remove!

File: src/main/resources/report_seed_data.json (hunk 5 above)
Comment (perrymant) [reply src/main/resources/report_seed_data.json:12]: Yep, my bad. Will get that in there.

**Pull request closed without merging**